### ISSUE TITLE
teamcity_build_config - Build Step Ordering

### ISSUE THREAD

**Issue by MattHodge** [cvbarros/terraform-provider-teamcity#43]
There is currently no way to control the build step ordering inside the `teamcity_build_config` resource.

Given the following Terraform:

```hcl
resource "teamcity_project" "step_ordering" {
  name = "Build Step Ordering"
}

resource "teamcity_build_config" "step_ordering" {
	name = "build config test"
	project_id = "${teamcity_project.step_ordering.id}"

	step {
		type = "cmd_line"
		name = "01 - First"
		code = "echo \"Hello First\""
	}

	step {
		type = "cmd_line"
		name = "02 - Second"
		code = "echo \"Hello Second\""
	}

	step {
		type = "cmd_line"
		name = "03 - Third"
		code = "echo \"Hello Third\""
	}
}
```

The steps are created in a different order:

![image](https://user-images.githubusercontent.com/1966555/50084300-cd551a80-01f6-11e9-882c-46c93e904d94.png)

**Comment by ss-remygreinhofer** [comment]
Would "simply" changing the schema to having a list of steps be right way to solve this issue?

**Comment by cvbarros** [comment]
This has been fixed locally, I'm delaying the push because it will be a breaking change for existing state, so I'm thinking how the best approach would be - to either start creating state migrations or just a hard compatibility break (which would adhere to current SemVer under 1.x).

**Comment by lmayorga1980** [comment]
This is a very nice to have. Currently I have different results during changes on the steps.

**Comment by augustgerro** [comment]
Any ideas on how to work around for now?

Let's say, I have three steps with:
1. build a docker container
2. push it 
3. deploy it

When steps go not one after one it's broke all builds.

**Comment by cvbarros** [comment]
Fixed via #70
